feature/home/src/main/java/com/android/developers/androidify/home/HomeScreenCompact.kt

```diff
@@ -0,0 +1,189 @@
+                MainHomeContent(
+                    dancingBotLink = dancingBotLink,
+                    modifier = Modifier
+                        .padding(horizontal = 16.dp)
+                        .align(Alignment.CenterHorizontally),
+                )
+            } else {
+                Box(modifier = Modifier.fillMaxSize()) {
+                    VideoPlayerRotatedCard(
+                        videoLink = videoLink,
+                        modifier = Modifier
+                            .padding(horizontal = 32.dp)
+                            .align(Alignment.Center),
+                    )
+                }
+            }
+        }
+        Row(
+            modifier = Modifier
+                .weight(.1f)
+                .fillMaxWidth(),
+            horizontalArrangement = Arrangement.Center,
+            verticalAlignment = Alignment.CenterVertically,
+        ) {
+            repeat(pagerState.pageCount) { iteration ->
+                val isCurrent by remember { derivedStateOf<Boolean> { pagerState.currentPage == iteration } }
+                val animatedColor by animateColorAsState(
+                    if (isCurrent) MaterialTheme.colorScheme.tertiary else MaterialTheme.colorScheme.onTertiary,
+                    label = "animatedFirstColor",
+                )
+
+                Box(
+                    modifier = Modifier
+                        .padding(2.dp)
+                        .clip(RoundedCornerShape(size = 16.dp))
+                        .animateContentSize()
+                        .background(
+                            color = animatedColor,
+                            shape = RoundedCornerShape(size = 16.dp),
+                        )
+                        .height(16.dp)
+                        .width(if (isCurrent) 40.dp else 16.dp),
+                )
+            }
+        }
+        Spacer(modifier = Modifier.size(12.dp))
+        var buttonPosition by remember {
+            mutableStateOf(IntOffset.Zero)
+        }
+        Column {
+            if (couldRequestFullSpace()) {
+                ViewInFullSpaceButton(
+                    modifier = Modifier
+                        .height(64.dp),
+                )
+                Spacer(modifier = Modifier.size(12.dp))
+            }
+            HomePageButton(
+                modifier = Modifier
+                    .onLayoutRectChanged {
+                        buttonPosition = it.boundsInWindow.center
+                    }
+                    .padding(bottom = 16.dp)
+                    .height(64.dp)
+                    .width(220.dp),
+                colors = ButtonDefaults.buttonColors()
+                    .copy(containerColor = Blue),
+                onClick = {
+                    onClick(buttonPosition)
+                },
+            )
+        }
+    }
+}
+
+@ExperimentalMaterial3ExpressiveApi
+@PhonePreview
+@Composable
+private fun HomeScreenPhonePreview() {
+    NoXrSupportPreview {
```

Comment:
I'd prefer if we could rather have this only when required for XR, as that'd be majority of use cases, and rather only add this if its an XR Preview, otherwise we are scattering XR logic around the app when its not required. Thinking of the case of if we ever remove support for XR, we'd need to go and remove all of these NoXrSupportPreview to the standard app previews.

Reply:
100% agree - I needed to add this due to a bug in Jetpack XR Compose - [b/441901724](b/441901724). In `alpha06`, any attempt to access and get the precomputed values of `LocalSession`, `LocalSpatialCapabilities`, `LocalSpatialConfiguration` in any child component in the Previewer results in an exception, causing the preview to fail entirely. In other words, even in a 2D layout, even checking for the existence of XR will cause the previewer to fail.

`NoXrSupportPreview` provides mock implementations for the `CompositionLocal` objects to work around this bug. I'm hoping to be able to remove these come `alpha07`.